src/bun.js/bindings/bindings.zig

```diff
@@ -4856,6 +4859,11 @@ pub const JSValue = enum(JSValueReprInt) {
         return FFI.JSVALUE_TO_INT32(.{ .asJSValue = this });
     }
 
+    pub fn asFileDescriptor(this: JSValue) bun.FileDescriptor {
+        std.debug.assert(this.isNumber());
+        return bun.FDImpl.fromUV(this.asInt32()).encode();
+    }
```

Comment:
is this lossy on windows? I see there's also the method `value.toInt64()`

Reply:
It's not lossy, but `isNumber` is both decimal and int32. `asInt32` is only int32. This would produce unexpected results when the value is a double. Instead, this should either use `coerce` or `toInt32`